tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -2077,15 +2263,561 @@ TEST(LinalgBackendEigen, SGMatrix_rank_update)
 	b[1] = 3;
 
 	auto A2 = A.clone();
-	Map<MatrixXd> A2_eig(A2.matrix, size, size);
+	Map<Mxx> A2_eig(A2.matrix, size, size);
 	A2(0, 0) += b[0] * b[0];
 	A2(0, 1) += b[0] * b[1];
 	A2(1, 0) += b[1] * b[0];
 	A2(1, 1) += b[1] * b[1];
 
-	rank_update(A, b);
-	EXPECT_NEAR((A2_eig - A_eig).norm(), 0.0, 1e-14);
+	rank_update(A, b, static_cast<ST>(1));
+	EXPECT_NEAR((A2_eig - A_eig).norm(), 0, get_epsilon<ST>());
+
+	rank_update(A, b, static_cast<ST>(-1));
+	EXPECT_NEAR((A_eig - A_eig).norm(), 0, get_epsilon<ST>());
+}
+
+// test types based on shogun/mathematics/linalg/LinalgBackendBase.h
+TYPED_TEST(LinalgBackendEigenAllTypesTest, SGVector_add)
+{
+	check_SGVector_add<TypeParam>();
```

Comment:
again the question: if I grep for `check_SGVector_add<`, I only find this occurrence of the call, why not move the code in here? Did I miss a response for this anywhere?

Reply:
I was just asking what was the motivation here: why put the code into a call rather than moving it into the test directly? If there is a good reason it is fine of course, but if there is none, then maybe we should move it :)

Reply:
ok then, let's keep it, no big deal

Reply:
actually, thought about it again, and I think it would be better to move the code down (for those instances where the function is not re-used anywhere else) to keep the spaghetti factor down a bit. These things tend to be annoying years in the future when refactoring stuff.

Reply:
OK, I'll do that!